__tests__/app.test.js

```diff
@@ -321,10 +325,55 @@ describe('/api/articles/:article_id/comments', () => {
       .expect(200)
       .then((res) => {
         res.body.forEach((user) => {
+          expect(res.body.length > 0).toBe(true)
           expect(user).hasOwnProperty("username");
           expect(user).hasOwnProperty("name");
           expect(user).hasOwnProperty("avatar_url");
         })
       })
     })
-  });
\ No newline at end of file
+  });
+
+  describe('GET /api/articles (topic query)', () => {
+    test('Returns an array of article objects with only the topics of mitch',() => {
+      return request(app)
+      .get('/api/articles/?topic=mitch')
+      .expect(200)
+      .then((res) => {
+        expect(Array.isArray(res.body)).toBe(true)
+        expect(res.body.length).toBe(12)
+        res.body.forEach((article) => {
+          expect(res.body.length > 0).toBe(true)
+          expect(article.topic).toBe('mitch');
+        })
+      })
+      })
+    test('returns an array of article objects with all topics included when no query is used', () => {
+      return request(app)
+      .get('/api/articles/')
+      .expect(200)
+      .then((res) => {
+        res.body.forEach((user) => {
+          expect(res.body.length > 0).toBe(true)
+          expect(user.topic).toMatch(/mitch|cats/) 
+        })
+      })  
+    })
+    test('returns an empty array when a valid topic is used but the topic isnt found on any article', () => {
+      return request(app)
+      .get('/api/articles/?topic=paper')
+      .expect(200)
+      .then((res) => {
+        expect(res.body.length).toBe(0)
+
+      })
+    })
+    test('returns an 400 when an invalid topic is used', () => {
```

Comment:
This is not an invalid topic, but a non-existent one, which should respond with 404, not found.